src/Client/index.ts

```diff
@@ -356,13 +364,51 @@ export class Client {
       const stream = await creator(client);
 
       cache?.set(client, stream);
-
-      return stream.on("error", (err) => {
-        cache?.delete(client);
-        this.handleError(client, err);
-      });
+      this.disposableStreams.add(stream);
+
+      return stream
+        .on("error", (err) => {
+          cache?.delete(client);
+          this.disposableStreams.delete(stream);
+          this.handleError(client, err);
+        })
+        .on("end", () => {
```

Comment:
**CONCERN:** Shouldn't we also do the same for `closed` event?

Reply:
Very good spot, was working as all the streams are actually duplex, so emit an `end` event.

Looks like `finished` is the correct way to handle this:
https://nodejs.org/api/stream.html#streamfinishedstream-options-callback